src/page/login/login.tsx

```diff
@@ -1,100 +1,112 @@
-import React from 'react';
+import React, {useState, useEffect} from 'react';
 import {
   View,
   Text,
   TextInput,
   TouchableHighlight,
   Keyboard,
   TouchableOpacity,
+  ActivityIndicator,
 } from 'react-native';
 import {styles} from './loginCss';
 import Reactotron from 'reactotron-react-native';
-// import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
-// import {faCoffee} from '@fortawesome/free-solid-svg-icons';
-// import Logo from '../../resource/images/logo.svg';
+import {useNavigation, StackRouter} from '@react-navigation/native';
+import {connect, ConnectedProps} from 'react-redux';
+import {setNavigation, clearSetting} from '../../store/login/saga';
+import LinearGradient from 'react-native-linear-gradient';
+import {
+  updateAccount,
+  updatePassword,
+  fetchData,
+} from '../../store/login/action';
+
+const Login = (props: PropsFromRedux) => {
+  const navitation = useNavigation();
+  const {dispatch, loading} = props;
+  useEffect(() => {
+    setNavigation(navitation);
+    return clearSetting;
+  }, [navitation]);
 
-const Login: React.FC = () => {
   /**
    * login
    */
   function login() {
-    //https://reactnative.dev/movies.json
-    return (
-      fetch('https://reactnative.dev/movies.json', {
-        // method: 'POST',
-        // headers: {
-        //   Accept: 'application/json',
-        //   'Content-Type': 'application/json',
-        // },
-        // body: {
-        //   loginname: 'jerry_liu',
-        //   password: '1qaz@WSX',
-        //   callback: () => {
-        //     Reactotron.log('123');
-        //   },
-        // },
-      })
-        .then(response => {
-          Reactotron.log(response);
-          response.json();
-        })
-        // .then(json => {
-        //   Reactotron.log(json);
-        //   return json.movies;
-        // })
-        .catch(error => {
-          console.error(error);
-        })
-    );
+    // navitation.navigate('Main');
+    dispatch(fetchData(navitation));
```

Comment:
Please never pass navigation through the actions. It may lead to memory leaks.
Instead, create Navigation service like [here](https://reactnavigation.org/docs/navigating-without-navigation-prop) described.